src/js/terms-and-conditions/containers/MhvTermsAndConditions.jsx

```diff
@@ -0,0 +1,216 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import Scroll from 'react-scroll';
+import appendQuery from 'append-query';
+import URLSearchParams from 'url-search-params';
+
+import AlertBox from '@department-of-veterans-affairs/jean-pants/AlertBox';
+import LoadingIndicator from '@department-of-veterans-affairs/jean-pants/LoadingIndicator';
+
+import { getScrollOptions } from '../../../platform/utilities/ui';
+
+import {
+  acceptTerms,
+  fetchLatestTerms,
+  fetchTermsAcceptance
+} from '../actions';
+
+const ScrollElement = Scroll.Element;
+const scroller = Scroll.scroller;
+
+const TERMS_NAME = 'mhvac';
+
+const unagreedBannerProps = {
+  headline: 'Using Vets.gov Health Tools',
+  content: 'Before you can use the health tools on Vets.gov, you’ll need to read and agree to the Terms and Conditions below. This will give us your permission to show you your VA medical information on this site.'
+};
+
+export class MhvTermsAndConditions extends React.Component {
+  constructor(props) {
+    super(props);
+
+    const searchParams = new URLSearchParams(window.location.search);
+    this.redirectUrl = searchParams.get('tc_redirect');
+
+    this.state = {
+      isAgreementChecked: false,
+      showAcceptedMessage: false,
+      showCanceledMessage: false
+    };
+  }
+
+  componentDidMount() {
+    this.props.fetchLatestTerms(TERMS_NAME);
+    if (sessionStorage.userToken) { this.props.fetchTermsAcceptance(TERMS_NAME); }
+  }
+
+  redirect = () =>  {
+    if (this.redirectUrl) {
+      const newUrl = appendQuery(this.redirectUrl, { tc_accepted: true }); // eslint-disable-line camelcase
+      window.location.replace(newUrl);
+    }
+  }
+
+  handleAgreementCheck = () => {
+    this.setState({ isAgreementChecked: !this.state.isAgreementChecked });
+  }
+
+  handleSubmit = (e) => {
+    e.preventDefault();
+    this.props.acceptTerms(TERMS_NAME);
+    this.setState({ showAcceptedMessage: true }, () => {
+      scroller.scrollTo('banner', getScrollOptions());
+      this.redirect();
+    });
+  }
+
+  handleCancel = (e) => {
+    e.preventDefault();
+    this.setState({ showCanceledMessage: true }, () => {
+      scroller.scrollTo('banner', getScrollOptions());
+    });
+  }
+
+  handleCloseBanner = () => {
+    this.setState({
+      showAcceptedMessage: false,
+      showCanceledMessage: false
+    });
+  }
+
+  renderBanner = () => {
+    let bannerProps = null;
+
+    if (this.state.showAcceptedMessage) {
+      bannerProps = {
+        headline: 'You’ve accepted the Terms and Conditions for using Vets.gov health tools',
+        content: ''
+      };
+    } else if (this.state.showCanceledMessage) {
+      bannerProps = unagreedBannerProps;
+    }
+
+    return bannerProps && (
+      <ScrollElement name="banner">
+        <AlertBox
+          {...bannerProps}
+          isVisible
+          status="success"
+          onCloseAlert={this.handleCloseBanner}/>
```

Comment:
Should we add anything here to account for errors stored in `this.props.errors`? I see that added into the reducer but don't think it's being rendered.

Reply:
We probably should. Don't have design/copy for it yet, though.